repos/system_upgrade/el7toel8/actors/firewalldcheckebtablesbroute/actor.py

```diff
@@ -0,0 +1,36 @@
+from leapp.actors import Actor
+from leapp.models import Inhibitor
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.libraries.actor import private
+
+import os
+import xml.etree.ElementTree as ElementTree
+
+
+class FirewalldCheckEbtablesBroute(Actor):
+    """
+    In RHEL-8 the ebtables table broute is not available. If the user is using
+    firewalld direct rules that utilize this table then we need to inhibit the
+    upgrade.
+    """
+
+    name = 'firewalld_check_ebtables_broute'
+    consumes = ()
+    produces = (Inhibitor,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        path = '/etc/firewalld/direct.xml'
+        if not os.path.exists(path):
+            return
+
+        tree = ElementTree.parse(path)
+        root = tree.getroot()
```

Comment:
you cannot interact with system in `ChecksPhase`. See the [IPU workflow](https://github.com/oamg/leapp-repository/blob/a72e1c0d311dfbcb0121db79305b9a0e48474dc4/repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py#L18) file with description of all phases. When you want to read a file from the system, you need to create actor in scanner phase and produce message using a model. So the file will be read only once and other actors will use the data.

And to inhibit the upgrade, you need an actor running in `ChecksPhase` which will consume those data.